zerver/tests/test_custom_profile_data.py

```diff
@@ -600,6 +600,36 @@ def test_update(self) -> None:
             result, "Only 2 custom profile fields can be displayed in the profile summary."
         )
 
+    def test_update_name_and_hint_optional(self) -> None:
+        self.login("iago")
```

Comment:
I am not sure whether these tests are needed. We can probably update existing tests to just pass the parameter that is updated in the test like we do for many other tests.

We have other `PATCH` endpoints as well where we do not pass all the parameters in request but we do not have such separate tests checking what happens if not pass in a certain parameter.

Reply:
I have removed name and hint fields in tests where other fields were being tested but name and hint were passed so as not to fail the tests. I have kept the tests to make sure that the display_in_profile_summary and required do not become false if they are passed as None, previously that was the case with display_in_profile_summary and there was no test for it. 

Changes at https://github.com/zulip/zulip/compare/34ddac6155c722de7484f216af0ef11ab42a3b00..c4c980bb5745e07ce93431b99e8d9b0a7cc9ef5d